pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -135,34 +138,91 @@ func (b *AccessLogBuilder) setTCPAccessLog(mesh *meshconfig.MeshConfig, config *
 	}
 }
 
-func (b *AccessLogBuilder) setHTTPAccessLog(mesh *meshconfig.MeshConfig, connectionManager *hcm.HttpConnectionManager) {
-	if mesh.AccessLogFile != "" {
-		connectionManager.AccessLog = append(connectionManager.AccessLog, b.buildFileAccessLog(mesh))
+func buildAccessLogFromTelemetry(mesh *meshconfig.MeshConfig, spec *telemetry.Telemetry, forListener bool) *accesslog.AccessLog {
+	// TODO support multiple
+	accessLogConfig := spec.AccessLogging[0]
+	if accessLogConfig.GetDisabled().GetValue() {
+		return nil
 	}
 
-	if mesh.EnableEnvoyAccessLogService {
-		connectionManager.AccessLog = append(connectionManager.AccessLog, b.httpGrpcAccessLog)
+	// provider config
+	var providerName string
+	if len(mesh.GetDefaultProviders().GetAccessLogging()) > 0 {
+		providerName = mesh.GetDefaultProviders().GetAccessLogging()[0]
+	}
+	if len(accessLogConfig.Providers) > 0 {
+		// only one provider is currently supported, safe to take first
+		providerName = accessLogConfig.Providers[0].Name
+	}
+
+	for _, p := range mesh.ExtensionProviders {
+		if strings.EqualFold(p.Name, providerName) {
+			switch prov := p.Provider.(type) {
+			case *meshconfig.MeshConfig_ExtensionProvider_EnvoyFileAccessLog:
+				al := buildFileAccessLogHelper(prov.EnvoyFileAccessLog.Path, mesh)
+				if forListener {
+					al.Filter = addAccessLogFilter()
+				}
+				return al
+			default:
+				log.Debugf("unsupported access log provider %v: %T", providerName, prov)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func (b *AccessLogBuilder) setHTTPAccessLog(opts buildListenerOpts, connectionManager *hcm.HttpConnectionManager) {
+	mesh := opts.push.Mesh
+	spec := opts.push.Telemetry.EffectiveTelemetry(opts.proxy)
+
+	if len(spec.GetAccessLogging()) == 0 {
+		// No Telemetry API configured, fall back to legacy mesh config setting
+		if mesh.AccessLogFile != "" {
+			connectionManager.AccessLog = append(connectionManager.AccessLog, b.buildFileAccessLog(mesh))
+		}
+
+		if mesh.EnableEnvoyAccessLogService {
+			connectionManager.AccessLog = append(connectionManager.AccessLog, b.httpGrpcAccessLog)
+		}
+		return
+	}
+
+	if al := buildAccessLogFromTelemetry(mesh, spec, false); al != nil {
+		connectionManager.AccessLog = append(connectionManager.AccessLog, al)
 	}
 }
 
-func (b *AccessLogBuilder) setListenerAccessLog(mesh *meshconfig.MeshConfig, listener *listener.Listener) {
+func (b *AccessLogBuilder) setListenerAccessLog(push *model.PushContext, proxy *model.Proxy, listener *listener.Listener) {
+	mesh := push.Mesh
+	spec := push.Telemetry.EffectiveTelemetry(proxy)
 	if mesh.DisableEnvoyListenerLog {
 		return
 	}
-	if mesh.AccessLogFile != "" {
-		listener.AccessLog = append(listener.AccessLog, b.buildListenerFileAccessLog(mesh))
+
+	if len(spec.GetAccessLogging()) == 0 {
```

Comment:
This means if custom access log is set by telemetry api, the grpc access log is disabled.

If this is intended, i think the telemetry api is not complete.

Reply:
That is correct, the Telemetry API is not complete yet. It will be added in the future